### PR TITLE
Bugfix/2098 allows null values when nullable

### LINKED ISSUE #2098: Enum validation should not flag null values in nullable enums
### Q&A (please complete the following information)
 - OS: Windows 10
 - Browser: Chrome latest
 - Method of installation: https://editor.swagger.io
 - Swagger-Editor version: 3.8.0
 - Swagger/OpenAPI version: OpenAPI 3.0

### Content & configuration
Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.2
info:
  version: 1.0.0
  title: Nullable enum
paths: {}
components:
  schemas:
    Myschema:
      type: object
      properties:
        nullableProp:
          type: string
          nullable: true
          enum:
            - null
            - 'value1'
            - 'value2'
```

### Describe the bug you're encountering
This is a follow-up to PR #2079.

According to OpenAPI Specification maintainers ([1](https://github.com/OAI/OpenAPI-Specification/blob/master/proposals/003_Clarify-Nullable.md#if-a-schema-specifies-nullable-true-and-enum-1-2-3-does-that-schema-allow-null-values-see-1900), [2](https://github.com/OAI/OpenAPI-Specification/issues/1900#issuecomment-578409376)), nullable enums are defined as shown above. That is, the schema should have `nullable: true` AND also include `null` (not `"null"`) in the list of enum values.

The issue is that the editor flags the `null` value and says "enum value should conform to its schema's type". But in this example, `null` does conform to the schema type, because `nullable: true` acts as a `type` modifier.

### Expected behavior
`null` in the definition above should NOT trigger warnings.

### PR DESCRIPTION
**Pull request by ralphdoe**
### Description

I have added a validation for nullable properties. 

### Motivation and Context

Fixes #2098 

### How Has This Been Tested?

Added a Test case for this. 

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/plugins/validate-semantic/validators/2and3/schemas.js [hunk 1]
```diff
@@ -23,41 +23,46 @@ export const validate2And3TypeArrayRequiresItems = () => (system) => {
...
+
+
 export const validate2And3TypesInDefaultValuesMatchesWithEnum = () => (system) => {
   return system.validateSelectors
     .allSchemas()
     .then(nodes => {
       return nodes.reduce((acc, node) => {
         const schemaObj = node.node
         const { type } = schemaObj || {}
+        const isNullable = schemaObj.nullable ? schemaObj.nullable : false
```
Comment (shockey) [inline src/plugins/validate-semantic/validators/2and3/schemas.js]:
```suggestion
        const isNullable = !!schemaObj.nullable
```

since `nullable` should always be a boolean, we can get away with coercing it directly!

File: src/plugins/validate-semantic/validators/2and3/schemas.js [hunk 2]
```diff
@@ -23,41 +23,46 @@ export const validate2And3TypeArrayRequiresItems = () => (system) => {
...
-              isValidFormat = false
-              enumIndex = index
-            }
-            if (!isValidFormat) {
-              acc.push({
-                message: "enum value should conform to its schema's `type`",
-                path: [...node.path, "enum", enumIndex],
-                level: "warning",
-              })
+            if ((element === null & !isNullable) || element !== null) {
```
Comment (shockey) [inline src/plugins/validate-semantic/validators/2and3/schemas.js]: This conditional is a bit confusing to follow, and adds to the already-not-great indentation level in this validator.

You can achieve equivalent functionality by returning from the `forEach` iterator function:

```suggestion
            if (element === null && isNullable) {
              return
            }
```

This pattern is common in JavaScript, and is roughly equivalent to a `continue` statement in a conventional loop:

```js
const arr = [1, 2, 3, 4, 5]

// both of the loops below log the same four lines: "1", "3", "4", "5"

for(var num of arr) {
  if(num === 2) {
    continue
  }
  console.log(num)
}

arr.forEach(num => {
  if(num === 2) {
    return
  }
  console.log(num)
}
```

(if you commit this, don't forget to pull in the suggestion to delete the closing bracket below!)

File: src/plugins/validate-semantic/validators/2and3/schemas.js [hunk 3]
```diff
@@ -23,41 +23,46 @@ export const validate2And3TypeArrayRequiresItems = () => (system) => {
...
+                enumIndex = index
+              }
+              if (!isValidFormat) {
+                acc.push({
+                  message: "enum value should conform to its schema's `type`",
+                  path: [...node.path, "enum", enumIndex],
+                  level: "warning",
+                })
+              }
             }
```
Comment (shockey) [inline src/plugins/validate-semantic/validators/2and3/schemas.js]:
```suggestion
```

File: src/plugins/validate-semantic/validators/2and3/schemas.js (hunk 1 above)
Comment (ralphdoe) [reply src/plugins/validate-semantic/validators/2and3/schemas.js]: Done! Good advice!

File: src/plugins/validate-semantic/validators/2and3/schemas.js (hunk 2 above)
Comment (ralphdoe) [reply src/plugins/validate-semantic/validators/2and3/schemas.js]: Sweeet! Thanks!

File: src/plugins/validate-semantic/validators/2and3/schemas.js (hunk 3 above)
Comment (ralphdoe) [reply src/plugins/validate-semantic/validators/2and3/schemas.js]: Done, deleted!

**Comment by shockey** [comment]
:shipit: 

thanks @ralphdoe!
